subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyAdder.java

```diff
@@ -173,4 +174,56 @@ public interface DependencyAdder {
      */
     <D extends Dependency> void bundle(ProviderConvertible<? extends Iterable<? extends D>> bundle, Action<? super D> configuration);
 
+    /**
+     * Add a plugin dependency, specified by id.
+     *
+     * @param id the id of the dependency
+     * @since 8.1
+     */
+    void plugin(String id);
```

Comment:
This will result in e.g. `implementation.plugin("id")`, not `implementation(plugin("id"))`. I think the latter is more "Gradle-y", the only thing that does not follow those rules is `implementation.bundle` because it is multiple dependencies, not a single dependency.

To do `implementation(plugin("id"))`, a new creation method should be added to `DependencyFactory`, and `Dependencies` should be modified to use it with a `ExternalModuleDependency plugin(String id)` method. No `Provider` version should be added to `Dependencies` because the return type would conflict. Though, I think a `Provider<? extends ExternalModuleDependency> plugin(Provider<? extends PluginDependency> provider)` may also be appropriate, to handle version catalog.

Reply:
We thought about that but went with making it similar to `implementation.bundle(...)` at first.

Reply:
Thanks for the suggestion, I agree that we should make the test-suites dependencies block consistent with the top-level. I'll update accordingly.